client/finality-grandpa/src/communication/gossip.rs

```diff
@@ -1211,12 +1235,27 @@ impl<Block: BlockT> GossipValidator<Block> {
 	pub(super) fn new(
 		config: crate::Config,
 		set_state: environment::SharedVoterSetState<Block>,
+		prometheus_registry: Option<&Registry>,
 	) -> (GossipValidator<Block>, mpsc::UnboundedReceiver<PeerReport>)	{
+		let metrics = match prometheus_registry {
```

Comment:
```suggestion
		let metrics = match prometheus_registry.map(Metrics::register) {
```

Could reduce the code a little bit.